src/screens/ProfileScreen/ProfileScreen.js

```diff
@@ -1,21 +1,234 @@
-import React, { Component } from 'react';
-import { View, Text, TouchableOpacity } from 'react-native';
-import styles from './styles';
+import React, { useState, useEffect } from 'react';
+import { View, Text, TextInput, TouchableOpacity, SafeAreaView, Image, ScrollView, Platform } from 'react-native';
 import { firebase } from '../../config/config';
+import ImagePicker from 'react-native-image-crop-picker';
+import styles from './styles';
+import { LOGO, NO_IMAGE } from '../../config/routes.js';
+
+const ProfileScreen = () => {
+
+    const [name, setName] = useState("")
+    const [occupation, setOccupation] = useState("")
+    const [interest, setInterest] = useState("")
+    const [country, setCountry] = useState("")
+    const [editfield, setEditfield] = useState(false)
+    const [image, setImage] = useState(null)
+    const [displayImage, setDisplayImage] = useState(null);
+
+    const userID1 = firebase.auth().currentUser.uid
+    const ref = firebase.firestore().collection("users").doc(userID1);
+
+    useEffect(() => {
+        ref.get().then((doc) => {
+            if (doc.exists) {
+                setName(doc.data().fullName);
+                setInterest(doc.data().interest)
+                setCountry(doc.data().country)
+                setOccupation(doc.data().occupation)
+                setDisplayImage(doc.data().downloadURL)
+            } else {
+              alert("No Document Found")
+            }
+        }).catch((error) => {
+            alert(error)
+        });     
+  }, [])
+
+    const uploadImage = async () => {
+      if(image == null){
+        savePostData(displayImage) 
+        return
+      }
+        const uri = image;
+        const childPath = `post/${firebase.auth().currentUser.uid}/${Math.random().toString(36)}`;
+
+        const response = await fetch(uri);
+        const blob = await response.blob();
+
+        const task = firebase
+            .storage()
+            .ref()
+            .child(childPath)
+            .put(blob);
+
+        const taskProgress = snapshot => {
+            // progress tracking purposes only.
+            // use console for tracking purpose.
+        }
 
-class ProfileScreen extends Component{
-    render(){
-        return(
-            <View style={{flex:1, justifyContent: 'center'}}>
-               <View style={{alignSelf: 'center', width : '90%', marginTop: 100}}>
-                    <Text style={{alignSelf: 'center'}}>Profile Screen</Text>
-                    <TouchableOpacity style= {styles.LoginButtonView} onPress={() => firebase.auth().signOut()}>
-                        <Text style={styles.LoginButtonText}>Logout</Text>
+        const taskCompleted = () => {
+            task.snapshot.ref.getDownloadURL().then((snapshot) => {
+                savePostData(snapshot);
+                alert("Image Uploaded Succesfully")
```

Comment:
This is fine in this case. Better to have a different component for show messages by customizing it. Let's open a separate issue to create a component for show messages. That component should be customizable by parameters(such like the message showing UI should be different for errors and success cases).

Reply:
Also can you update the PR with a gif how these error messages are shown once that happens?

Reply:
Okay! So that new Component first needs to be pushed in Go-Social Repository?

Reply:
I'll create an issue in Go-Social Repository, and once it got approve I'll replace it with the existing one! is it fine?

Reply:
In this Screen there were lots of alerts, so to overcome this I have used the progress bar. Here is this gif. Let me know if this is fine.

<img src="https://user-images.githubusercontent.com/40908684/121808844-0b244400-cc78-11eb-935d-0ad932aef858.gif" width="200" >

Reply:
Good job. This is what we need. One thing, is it possible to change the 'pick image' status to 'uploading' once the user has picked the image and the uploading is in progress? That would be much convenient way to do this.

Reply:
Done!